scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env -S python3 -B
+
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import subprocess
+import sys
+import time
+
+import click
+
+# Configure logging format.
+logging.basicConfig(level=logging.DEBUG, format='%(levelname)s - %(message)s')
+
+LINUX_TV_APP_LOGS = './scripts/tests/Linux-tv-app-logs.txt'
+LINUX_TV_CASTING_APP_LOGS = './scripts/tests/Linux-tv-casting-app-logs.txt'
+
+RUN_INTERVAL = 2
+
+VENDOR_ID = 65521
+PRODUCT_ID = 32769
+DEVICE_TYPE = 35
+
+
+# Dump the logs to the console in the case of an error.
+def dump_logs_to_console(log_file):
+
+    if log_file == LINUX_TV_CASTING_APP_LOGS:
+        logging.debug('Dumping Linux TV Casting App Logs to Console.')
+    elif log_file == LINUX_TV_APP_LOGS:
+        logging.debug('Dumping Linux TV App Logs to Console.')
+
+    with open(log_file, 'r') as file:
+        logs = file.read()
+        logging.debug(logs)
+
+
+# Remove the log files once the script is done running.
+def remove_log_file(log_file):
+
+    if os.path.exists(log_file):
+        os.remove(log_file)
+    else:
+        logging.error('The file does not exist.')
+
+
+# Whenever a failure is discovered, we should print 'Discovery failed!',
+# dump the logs, clean up the log files, exit on error.
+def handle_discovery_failure():
+
+    logging.error('Discovery failed!\n')
+
+    dump_logs_to_console(LINUX_TV_CASTING_APP_LOGS)
+    dump_logs_to_console(LINUX_TV_APP_LOGS)
+
+    remove_log_file(LINUX_TV_CASTING_APP_LOGS)
+    remove_log_file(LINUX_TV_APP_LOGS)
+
+    sys.exit(1)
+
+
+# Helper function to extract the integer value from a string.
+def extract_value_from_string(line):
+
+    value = line.split(':')[-1].strip().replace('\x1b[0m', '')
```

Comment:
Give examples in comments of expected inputs and outputs.
I am unclear what `line` may contain ... it seems to be `<anything>:<maybe-some-color-code>number` ... however this is hard to understand here. Why do we even have the color encodings here?

This needs examples and explanations

Reply:
This is an example output from the Linux tv-casting-app which will be the provided string input (`line`):
`\x1b[0;34m[1713741926895] [7276:9521344] [DIS] Vendor ID: 65521\x1b[0m`

The value to be extracted from the string is `65521`.

I have added the relevant comments in the latest commit to address this.